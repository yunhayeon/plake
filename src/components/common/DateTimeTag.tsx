import clsx from "clsx";
import dayjs from "dayjs";

import { cn } from "@/lib/utils";

interface IDateTimeTagProps {
  size?: "small" | "medium";
  date: dayjs.Dayjs;
}

const baseStyle = (size: IDateTimeTagProps["size"]) =>
  cn(
    "flex items-center justify-center rounded-[4px] bg-purple-200 py-[2px] text-sm font-medium",
    size === "small" && "px-[6px] text-xs",
    size === "medium" && "px-2 text-sm",
  );

const DateTimeTag = ({ size = "medium", date }: IDateTimeTagProps) => {
  const dateText = date.format("M월 DD일");
  const timeText = date.format("HH:mm");

  return (
    <div className="flex gap-2">
      <time aria-label="날짜" className={clsx("text-black", baseStyle(size))}>
        {dateText}
      </time>
      <time
        aria-label="시간"
        className={clsx("text-purple-600", baseStyle(size))}
      >
        {timeText}
      </time>
    </div>
  );
};

export default DateTimeTag;
