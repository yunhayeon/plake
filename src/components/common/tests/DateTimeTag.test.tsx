import { render, screen } from "@testing-library/react";
import dayjs from "dayjs";

import DateTimeTag from "@/components/common/DateTimeTag";

describe("DateTimeTag 컴포넌트 테스트", () => {
  const mockDate = dayjs("2024-03-10T12:34:56");

  it("date prop에 날짜를 지정하면 날짜와 시간을 표시한다.", () => {
    render(<DateTimeTag date={mockDate} />);

    expect(screen.getByText("3월 10일")).toBeInTheDocument();
    expect(screen.getByText("12:34")).toBeInTheDocument();
  });

  it("size가 small일 때 올바른 크기가 적용된다.", () => {
    render(<DateTimeTag size="small" date={mockDate} />);

    expect(screen.getByText("3월 10일")).toMatchSnapshot();
    expect(screen.getByText("12:34")).toMatchSnapshot();
  });

  it("size가 medium일 때 올바른 크기가 적용된다. ", () => {
    render(<DateTimeTag size="medium" date={mockDate} />);

    expect(screen.getByText("3월 10일")).toMatchSnapshot();
    expect(screen.getByText("12:34")).toMatchSnapshot();
  });
});
