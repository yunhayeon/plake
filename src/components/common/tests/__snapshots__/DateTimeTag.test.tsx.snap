// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeTag 컴포넌트 테스트 size가 medium일 때 올바른 크기가 적용된다.  1`] = `
<time
  aria-label="날짜"
  class="text-black flex items-center justify-center rounded-[4px] bg-purple-200 py-[2px] font-medium px-2 text-sm"
>
  3월 10일
</time>
`;

exports[`DateTimeTag 컴포넌트 테스트 size가 medium일 때 올바른 크기가 적용된다.  2`] = `
<time
  aria-label="시간"
  class="text-purple-600 flex items-center justify-center rounded-[4px] bg-purple-200 py-[2px] font-medium px-2 text-sm"
>
  12:34
</time>
`;

exports[`DateTimeTag 컴포넌트 테스트 size가 small일 때 올바른 크기가 적용된다. 1`] = `
<time
  aria-label="날짜"
  class="text-black flex items-center justify-center rounded-[4px] bg-purple-200 py-[2px] font-medium px-[6px] text-xs"
>
  3월 10일
</time>
`;

exports[`DateTimeTag 컴포넌트 테스트 size가 small일 때 올바른 크기가 적용된다. 2`] = `
<time
  aria-label="시간"
  class="text-purple-600 flex items-center justify-center rounded-[4px] bg-purple-200 py-[2px] font-medium px-[6px] text-xs"
>
  12:34
</time>
`;
